src/main/java/pl/psnc/dei/service/EuropeanaAnnotationsService.java

```diff
@@ -95,10 +97,11 @@ public String postTranscription(Transcription transcription) {
      * @return
      */
     public String updateTranscription(Transcription transcription) {
+        JsonObject convertedTranscription = TranscriptionConverter.convert(transcription.getTranscriptionContent());
         String annotationResponse = webClient.put()
-                .uri(b -> b.path(annotationApiEndpoint + (annotationApiEndpoint.endsWith("/") ? "" : "/") + transcription.getAnnotationId()).build())
+                .uri(b -> b.path((annotationApiEndpoint.endsWith("/") ? "" : "/") + transcription.getAnnotationId()).build())
```

Comment:
I am sure, this was one of legacy bug of `UpdateTask`. Old version generated links like `https://github.com/https://github.com/EnrichEuropeana/dei/pull/164`. In fact host address is set in `EuropeanaAnnotationService::configure`, which calls this method
```java
    protected void setRootUri(String rootUri) {
        URI uri = validateUri(rootUri);
        if (webClient != null) {
            DefaultUriBuilderFactory uriBuilderFactory = new DefaultUriBuilderFactory(uri.toString());
            uriBuilderFactory.setEncodingMode(DefaultUriBuilderFactory.EncodingMode.VALUES_ONLY);
            webClient = webClient.mutate()
                    .baseUrl(uri.toString())
                    .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
                    .defaultHeader(HttpHeaders.ACCEPT, MediaType.ALL_VALUE)
                    .uriBuilderFactory(uriBuilderFactory)
                    .build();
        }
    }
```

Reply:
If this is the case then how posting new annotations works correctly?

Reply:
So, there is difference in way we construct request. In case of `EuropeanaAnnotationService::postTranscription` we configure `webClient` with `.uri(String uri)` which `Specify the URI using an absolute, fully constructed URI.` [docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/reactive/function/client/WebClient.UriSpec.html)
```java
String annotationResponse = webClient.post()
                .uri(annotationApiEndpoint)
                .header("Authorization", "Bearer " + userToken)
...
```

On the other hand in `EuropeanaAnnotationService::updateTranscription` we use overload `uri(Function<UriBuilder,URI> uriFunction)` which `Specify the URI by through a UriBuilder`. Builder we use looks like this `b -> b.path(...).build()`, where `.path(...)` is documented as `Append to the path of this builder.` [docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/util/UriBuilder.html)
```java
String annotationResponse = webClient.put()
                .uri(b -> b.path((annotationApiEndpoint.endsWith("/") ? "" : "/") + transcription.getAnnotationId()).build())
                .header("Authorization", "Bearer " + userToken)
```

Reply:
If you are sure about it then ok...I was just wondering how can this work because during debug I saw that value of annotationApiEndpoint was just /api/annotation and not the whole URL starting with https and so on.

Reply:
But actually you may be right because digging deeper into properties that property seems to be the whole valid URL

Reply:
> If you are sure about it then ok...I was just wondering how can this work because during debug I saw that value of annotationApiEndpoint was just /api/annotation and not the whole URL starting with https and so on.

ah, yes, `webClient` debug output works like this - shows only endpoint, not host
```bash
PUT /annotation/xxxx HTTP/1.1
``` 
is perfectly fine